src/QsCompiler/CompilationManager/DataStructures.cs

```diff
@@ -41,69 +50,374 @@ internal class CodeLine
 
         internal readonly int Indentation; // Note: This denotes the initial indentation at the beginning of the line
         internal readonly ImmutableArray<int> ExcessBracketPositions;
+        internal readonly ImmutableArray<int> ErrorDelimiterPositions;
 
         // convention: first one opens a string, and we have the usual inclusions: {0,5} means the chars 0..4 are the content of a string
         // -> i.e. -1 means the string starts on a previous line, and an end delimiter that is equal to Text.Length means that the string continues on the next line
-        internal readonly ImmutableArray<int> StringDelimiters; // note that this property is only properly initialized *after* knowing the sourrounding lines
+        internal readonly ImmutableArray<int> StringDelimiters; // note that this property is only properly initialized *after* knowing the surrounding lines
```

Comment:
Why do you need to keep these positions in the object?

Reply:
These are used in the parser code.

Reply:
They are used later on in the parser logic. Since the CodeLines were previously storing these delimiters, I don't want to change that here and have to refactor a larger portion of the parser code than I strictly need to.